vpr/src/route/rr_graph_fpga_interchange.h

```diff
@@ -0,0 +1,82 @@
+/* Defines the function used to load an rr graph written in xml format into vpr*/
+
+#ifndef RR_GRAPH_FPGA_INTERCHANGE_H
+#define RR_GRAPH_FPGA_INTERCHANGE_H
+
+#include "rr_graph.h"
+#include "device_grid.h"
+#include "DeviceResources.capnp.h"
+#include "LogicalNetlist.capnp.h"
+#include "capnp/serialize.h"
+#include "capnp/serialize-packed.h"
+#include <functional>
+
+void build_rr_graph_fpga_interchange(const t_graph_type graph_type,
+                                     const DeviceGrid& grid,
+                                     const std::vector<t_segment_inf>& segment_inf,
+                                     const enum e_base_cost_type base_cost_type,
+                                     int* wire_to_rr_ipin_switch,
+                                     std::string& read_rr_graph_filename,
+                                     bool do_check_rr_graph);
+
```

Comment:
Add comment saying we need to hash tuples (and why) and that the following utility functions achieve that.